docs/user_guides/single_table/custom_constraints.rst

```diff
@@ -0,0 +1,236 @@
+
+- ``transform`` which is responsible for the forward pass when using ``transform`` strategy.
+- ``reverse_transform`` which defines how to reverse the transformation of the ``transform``.
+- ``is_valid`` which computes the invalid rows.
+
+Let's look at a demo dataset:
+
+.. ipython:: python
+    :okwarning:
+
+    from sdv.demo import load_tabular_demo
+
+    employees = load_tabular_demo()
+    employees
+
+The dataset defined in :ref:`_single_table_constraints` contains basic details about employees.
+We will use this dataset to demonstrate how you can create your own constraint. 
+
+
+Custom Constraints
+------------------
+
+Let's consider the following example where we wish to generate synthetic data and we want
+a particular column, age for example, to be even. We will define ``transform``, 
+``reverse_transform``, and ``is_valid`` methods to make our data satisfy our constraints.
+
+
+In this example, we would like to create a constraint that makes ``age`` even. We can
+create a ``transform`` function that makes all the ``age`` values even.
+
+.. ipython:: python
+    :okwarning:
+
+    def transform(table_data):
+        table_data['age'] = table_data['age'] * 2
+        return table_data
+
+We transform the column ``age`` to be even by multiplicating it with 2. We now define
+the ``reverse_transform`` as doing the opposite effect.
+
+.. ipython:: python
+    :okwarning:
+
+    def reverse_transform(table_data):
+        table_data['age'] = table_data['age'] / 2
+        return table_data
+
+Lastly, we write the ``is_valid`` function to assess whether the column is even or not.
+
+.. ipython:: python
+    :okwarning:
+
+    def is_valid(table_data):
+        return table_data['age'] % 2 == 0
+
+We put every thing together in ``CustomConstraint`` and then we can create our model and 
+generate synthetic data by passing the constraint we just created.
+
+.. ipython:: python
+    :okwarning:
+
+    from sdv.constraints import CustomConstraint
+    from sdv.tabular import GaussianCopula
+
+    constraint = CustomConstraint(
+        transform=transform, 
+        reverse_transform=reverse_transform, 
+        is_valid=is_valid
+    )
+
+    gc = GaussianCopula(constraints=[constraint])
+
+    gc.fit(employees)
+
+    sampled = gc.sample(10)
+
+When we view the ``sampled`` data, we should find that all the rows in the sampled data have an
+even age value.
+
+.. ipython:: python
+    :okwarning:
+
+    sampled
+
+
+.. note::
+    It is sufficient to define ``is_valid`` function alone. In this case, the constraint will
+    use ``reject_sampling`` strategy. For example, ``CustomConstraint(is_valid=is_valid)``.
+
+
+Applying Functions to Multiple Columns
+--------------------------------------
+
+What if I want to apply the even constraint to multiple columns? Say we want ``age``
+and ``age_when_joined`` to be both even. Rather than defining two constraints,
+we provide another style of writing functions such that the function should accept 
```

Comment:
`...two constraints, or editing the code of our functions for each new column that we want to constraint, we provide...`